src/Database/Esqueleto/Internal/Internal.hs

```diff
@@ -1371,7 +1372,11 @@ instance SqlString a => SqlString (Maybe a) where
 -- | Class that enables one to use 'toBaseId' to convert an entity's
 -- key on a query into another (cf. 'toBaseId').
 class ToBaseId ent where
+  -- | e.g. @type BaseEnt MyBase = MyChild@
   type BaseEnt ent :: *
+  -- | Convert from the key of the BaseEnt(ity) to the key of the child entity.
+  -- This function is not actually called, but that it typechecks proves this operation is safe.
+  -- You may be able to succinctly implement this with 'Data.Coerce.coerce'
```

Comment:
I think coerce is both a nice implementation tip, but also gives good intuition for what this function is about

Reply:
Hm, after thinking about it, it's probably a mistake to recommend `coerce`. Since:

```
ghci> :t coerce (UserKey UUID.nil) :: OrganizationId
coerce (UserKey UUID.nil) :: OrganizationId :: OrganizationId
```

is just fine, but is actually incorrect

Reply:
Yeah, `coerce` is a little unsafe, in that it *does* work but it relies on the user guaranteeing that the two underlying IDs are indeed the same.

Leaving it in there is probably fine but we should leave a warning to the edge case.